File: src/main/java/com/wangduwei/asm/copy/lsieun/asm/tree/ClassTransformer.java
package com.wangduwei.asm.copy.lsieun.asm.tree;

import org.objectweb.asm.tree.ClassNode;

public class ClassTransformer {
    protected ClassTransformer ct;

    public ClassTransformer(ClassTransformer ct) {
        this.ct = ct;
    }

    public void transform(ClassNode cn) {
        if (ct != null) {
            ct.transform(cn);
        }
    }
}

File: src/main/java/com/wangduwei/asm/source/part6/ClassTransformer.kt
package com.wangduwei.asm.source.part6

import org.objectweb.asm.tree.ClassNode

/**
 * @author 杜伟
 * @date 2021/8/11 8:48 PM
 *
 */
open class ClassTransformer(protected val ct:ClassTransformer? = null) {

    open fun transform(cn: ClassNode) {
        ct?.transform(cn)
    }
}